examples/room/lod/lod/manager.go

```diff
@@ -0,0 +1,289 @@
+				updatesSentCounter.WithLabelValues("far", "interval").Inc()
+				return true
+			}
+		}
+		updatesSkippedCounter.WithLabelValues("interval_not_met").Inc()
+	}
+
+	return false
+}
+
+// processSenderReceiverPair processes a single sender-receiver pair
+// Returns the receiver ID if a message should be sent, empty string otherwise
+func (m *Manager) processSenderReceiverPair(
+	senderID string,
+	senderEntity *UserEntity,
+	receiverID string,
+	receiverEntity *UserEntity,
+) string {
+	if senderID == receiverID {
+		updatesSkippedCounter.WithLabelValues("same_user").Inc()
+		return ""
+	}
+
+	distance := calculateDistance(senderEntity.X, senderEntity.Y, receiverEntity.X, receiverEntity.Y)
+	userDistancesHistogram.Observe(float64(distance))
+
+	shouldSend := m.shouldSendUpdate(senderID, senderEntity, receiverID, receiverEntity, distance)
+
+	if shouldSend {
+		return receiverID
+	}
+
+	return ""
+}
+
+// processSingleReceiver processes all senders for one receiver
+func (m *Manager) processSingleReceiver(
+	receiverUserID string,
+	receiverUserEntity *UserEntity,
+	userEntities map[string]*UserEntity,
+) {
+	for senderUserID, senderUserEntity := range userEntities {
+		targetID := m.processSenderReceiverPair(senderUserID, senderUserEntity, receiverUserID, receiverUserEntity)
+		if targetID != "" {
+			senderUserEntity.RememberMutex.Lock()
+			senderUserEntity.Remember[receiverUserID] = time.Now()
+			senderUserEntity.RememberMutex.Unlock()
+
+			select {
+			case m.sendBuffer <- sendMessage{
+				receiverUserID: receiverUserID,
+				ver:            m.ver,
+				cmd:            m.cmd,
+				payload:        senderUserEntity.Payload,
+			}:
+			default:
+				logger.Warnf("sendBuffer full, dropping message", "receiverUserID", receiverUserID, "senderUserID", senderUserID)
+				bufferDropsCounter.Inc()
+			}
+		}
+	}
+}
+
+// processAllUsers processes all users in the manager
+func (m *Manager) processAllUsers() {
+	start := time.Now()
+	m.managerMapMutex.RLock()
+	userEntities := maps.Clone(m.userEntities)
+	m.managerMapMutex.RUnlock()
+
+	for receiverID, receiverEntity := range userEntities {
+		m.processSingleReceiver(receiverID, receiverEntity, userEntities)
+	}
+
+	for _, userEntity := range userEntities {
+		userEntity.ChangedAfterSend = false
+	}
+
+	duration := time.Since(start)
+
+	loopDurationHistogram.Observe(duration.Seconds())
+	bufferSizeGauge.Set(int64(len(m.sendBuffer)))
+
+	logger.Debugf("invokeLodLoop", "time", duration)
+}
+
+// send packets to nearby users
+// 1. nearer than maxDistanceForNearby -> send in every syncIntervalForNearby
+// 2. farther than maxDistanceForFar -> don't send
+// 3. between maxDistanceForNearby and maxDistanceForFar -> send in every syncIntervalForFar
+func (m *Manager) invokeLodLoop() {
+	for {
+		if !m.started.Load() {
+			break
+		}
+		time.Sleep(time.Duration(m.syncIntervalForNearby) * time.Millisecond)
+		m.processAllUsers()
+	}
+	close(m.sendBuffer)
```

Comment:
IMO: I think it would make more sense for started to be a channel `done`
```suggestion
	t := time.NewTicker(time.Duration(m.syncIntervalForNearby) * time.Millisecond)
	for {
		select {
		case <-t.C:
			m.processAllUsers()
		case <-m.done:
			close(m.sendBuffer)
			return
		}
	}
```

Reply:
I agree, that make more sense. I'll change it.